ext/storage_sources/azure_store/test/test_azure_connection.cpp

```diff
@@ -30,5 +30,302 @@
+        objects.clear();
+        CHECK(conn.list_objects(prefix + file_name, objects, list_single) == 0);
+        CHECK(objects.size() == 1);
+
+        // There should be 2 matches with 'list_obj_1' prefix pattern.
+        objects.clear();
+        CHECK(conn.list_objects(prefix + file_name + "1", objects, !list_single) == 0);
+        CHECK(objects.size() == 2);
+
+        // Clean up.
+        for (int i = 0; i < num_objects; i++) {
+            auto blob_client =
+              azure_client.GetBlockBlobClient(prefix + file_name + std::to_string(i) + ".txt");
+            blob_client.Delete();
+        }
+    }
+
+    SECTION("(Testing delete functionality for Azure).", "[azure-connection]")
+    {
+        azure_connection conn = azure_connection("myblobcontainer1", "delete_obj_");
+
+        auto blob_client = azure_client.GetBlockBlobClient("delete_obj_" + file_name);
+
+        blob_client.UploadFrom(file_name);
+
+        // Test that an object can be deleted.
+        CHECK(conn.delete_object(file_name) == 0);
+
+        // Test that removing an object that doesn't exist throws an exception.
+        CHECK(conn.delete_object(non_exi_object_key) == ENOENT);
+    }
+
+    SECTION("(Checking put functionality in Azure)", "[azure-connection]")
+    {
+        azure_connection conn = azure_connection("myblobcontainer1", "put_obj_");
+
+        CHECK(conn.put_object(file_name, path) == 0);
+
+        auto blob_client = azure_client.GetBlockBlobClient("put_obj_" + file_name);
+        blob_client.Delete();
+
+        // Test that putting an object that doesn't exist locally throws an exception.
+        CHECK(conn.put_object(non_exi_object_key, non_exi_file_name) == -1);
+    }
+
+    SECTION("(Checking read functionality in Azure)", "[azure-connection]")
+    {
+        azure_connection conn = azure_connection("myblobcontainer1", "read_obj_");
+
+        auto blob_client = azure_client.GetBlockBlobClient("read_obj_" + file_name);
+
+        blob_client.UploadFrom(file_name);
+        void *buffer = calloc(1024, sizeof(char));
+
+        // Test reading whole file.
+        CHECK(conn.read_object(file_name, 0, payload.length(), buffer) == 0);
+        CHECK(static_cast<char *>(buffer) == payload);
+        memset(buffer, 0, 1024);
+
+        // Test overflow on positive offset but past EOF.
+        CHECK(conn.read_object(file_name, 1, 1000, buffer) == -1);
+        memset(buffer, 0, 1024);
+
+        // Test overflow on negative offset but past EOF.
+        CHECK(conn.read_object(file_name, -1, 1000, buffer) == -1);
+        memset(buffer, 0, 1024);
+
+        // Test overflow with negative offset.
+        CHECK(conn.read_object(file_name, -1, 12, buffer) == -1);
+        free(buffer);
+
+        blob_client.Delete();
+
+        // Tests that reading a non existant object returns an exception.
+        CHECK(conn.read_object(non_exi_object_key, 0, 1, buffer) == ENOENT);
+    }
+
+    SECTION("(Test complex case).", "[azure-connection]")
```

Comment:
The complex cases feel a bit repetitive. One large complex case with the connection class methods working dependently would be better

Maybe something like this?
Check non-existing object does not exist in the container
Put an object in the container
Read substring of existing object in the container
Put another object in the container
List object in the container, list single
Read an object in the container from middle to end
Put a few more objects in the container
List object in the container, list all
Check delete on non-existing object in the container
Check delete on existing object but wrong prefix
Check read on non-existing object in the container
Check read on existing object but wrong prefix
Read an existing object in the container with overflow
Check that an object put into the container exists in the container
Delete an object in the container
Check that the object no longer exists in the container
Clean up - delete all existing objects in the container

Reply:
resolved here https://github.com/wiredtiger/wiredtiger/pull/8747/commits/c412f5811a4ae070687ea915492f8bb4e10f51b8